components/ui/TextAreaInput.tsx

```diff
@@ -6,11 +6,14 @@ import {
   UseFormRegister,
 } from "react-hook-form";
 import { FormSchemaType as PatientInfoForm } from "../../pages/patient-information";
+import { FormSchemaType as CreateSubmissionForm } from "../../pages/submission/create";
 
 interface TextAreaInputProps {
   inputName: string;
   labelName: string;
-  register: UseFormRegister<PatientInfoForm>;
+  register:
+    | UseFormRegister<PatientInfoForm>
+    | UseFormRegister<CreateSubmissionForm>;
```

Comment:
ah, you need to use generics! otherwise as you see, you'll have to hardcode everything textarea will support, which doesn't make much sense. Lemme know if you need help setting this up